src/effects/ASCIIEffect.ts

```diff
@@ -0,0 +1,171 @@
+import { Color, Uniform, Vector4 } from "three";
+import { ASCIITexture } from "../textures/ASCIITexture.js";
+import { Effect } from "./Effect.js";
+import { AddBlendFunction } from "./blending/index.js";
+
+import fragmentShader from "./shaders/ascii.frag";
+
+/**
+ * ASCIIEffect options.
+ *
+ * @category Effects
+ */
+
+export interface ASCIIEffectOptions {
+  asciiTexture?: ASCIITexture;
+  cellSize?: number;
+  color?: Color;
+  inverted?: boolean;
+}
+
+/**
+ * An ASCII effect.
+ *
+ * @category Effects
+ */
+
+export class ASCIIEffect extends Effect {
+  /**
+   * @see {@link cellSize}
+   */
+
+  private _cellSize = -1;
```

Comment:
Class members should always be initialized in the constructor.

```suggestion
  private _cellSize: number;
```